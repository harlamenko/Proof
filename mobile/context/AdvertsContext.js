import createDataContext from "./createDataContext";
import ProofAPI from '../api/ProofAPI';
import { Advert } from "../models/Advert";
import Toast from 'react-native-simple-toast';

const initialFilter = {
  paging: {
    lastAdvert: null,
    seenIds: [],
    total: 0
  },
  search: {
    field: "publication_date",
    keyWords: null,
    direct: -1
  }
}

const advertsReducer = (prevState, action) => {
  switch (action.type) {
    case 'ADD_ADVERTS':
      return {
        ...prevState,
        adverts: [...prevState.adverts, ...action.payload]
      };
    case 'UPDATE_PAGING_STATE':
      const { lastAdvert, seenIds, total } = action.payload;

      return {
        ...prevState,
        paging: {
          seenIds: [...prevState.paging.seenIds, ...seenIds],
          lastAdvert,
          total
        }
      };
    case 'DROP_FILTER':
    case 'UPDATE_FILTER':
      return {
        ...action.payload,
        adverts: [],
        currentAdvert: null
      };
    case 'SET_EMPTY_MESSAGE':
      return {
        ...prevState,
        emptyMessage: action.payload
      };
    case 'SET_CURRENT_ADVERT':
      return {
        ...prevState,
        currentAdvert: action.payload
      }
    default:
      return prevState;
  }
};

const getAdverts = dispatch => async ({ paging, search }) => {
  try {
    const res = await ProofAPI.post('/adverts', { paging, search })
    const adverts = res.data.adverts.map(obj => new Advert(obj));

    if (!adverts.length) { return; }

    const { total } = res.data;
    const { publication_date, price } = adverts[adverts.length - 1];

    dispatch({ type: 'ADD_ADVERTS', payload: adverts });
    dispatch({
      type: 'UPDATE_PAGING_STATE',
      payload: {
        lastAdvert: { publication_date, price },
        seenIds: adverts.map(ad => ad.id),
        total
      }
    });
  } catch (err) {
    console.error(err);
  }
}

const updateFilter = dispatch => ({ search }) => {
  const { paging } = initialFilter;
  dispatch({ type: 'UPDATE_FILTER', payload: { search, paging } });
}

const dropFilter = dispatch => () => {
  dispatch({ type: 'DROP_FILTER', payload: initialFilter });
}

const getAdvertDetails = dispatch => async (id) => {
  try {
    const { data: advert } = await ProofAPI.get(`/adverts/${id}`);

    dispatch({ type: 'SET_CURRENT_ADVERT', payload: new Advert(advert) })
  } catch (err) {
    console.error(err);
  }
}

const getMyAdvert = dispatch => async uid => {
  try {
    const a = new Advert(uid);
    const { data: advert } = await ProofAPI.get(`/adverts/my/${a.build_id}`);
    if (advert) {
      dispatch({ type: 'SET_CURRENT_ADVERT', payload: new Advert(advert) })
    } else {
      dispatch({ type: 'SET_EMPTY_MESSAGE', payload: 'Ничего не найдено' })
    }
  } catch (err) {
    console.error(err);
  }
}

const clearEmptyMessage = dispatch => () => {
  dispatch({ type: 'SET_EMPTY_MESSAGE', payload: null })
}

const dropCurrentAdvert = dispatch => () => {
  dispatch({ type: 'SET_CURRENT_ADVERT', payload: null });
}

const setCurrentAdvert = dispatch => advert => {
  dispatch({ type: 'SET_CURRENT_ADVERT', payload: new Advert(advert) })
}

const deleteAdvert = dispatch => async id => {
  try {
    await ProofAPI.delete(`/adverts/${id}`);
    Toast.showWithGravity('Объявление успешно удалено!', Toast.SHORT, Toast.CENTER);

  } catch (err) {
    console.error(err);
    Toast.showWithGravity('Произошла ошибка', Toast.SHORT, Toast.CENTER);
  }

}

export const { Provider, Context } = createDataContext(
  advertsReducer,
  {
    getAdverts,
    updateFilter,
    dropFilter,
    getAdvertDetails,
    dropCurrentAdvert,
    setCurrentAdvert,
    deleteAdvert,
    getMyAdvert,
    clearEmptyMessage
  },
  { ...initialFilter, adverts: [], currentAdvert: null, emptyMessage: null }
);