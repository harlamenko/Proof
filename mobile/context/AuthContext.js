import { AsyncStorage } from 'react-native';
import Toast from 'react-native-simple-toast';
import ProofAPI from '../api/ProofAPI';
import createDataContext from './createDataContext';

export const initialState = {
  token: null,
  myAdvert: null,
  errorMessage: '',
  initialyLoaded: false,
  loading: false,
  registering: false,
  user: null,
};

export const CHANGE_REGISTERING = 'CHANGE_REGISTERING';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
export const SIGN_IN = 'SIGN_IN';
export const SIGN_OUT = 'SIGN_OUT';
export const SET_LOCAL_INFO = 'SET_LOCAL_INFO';
export const CLEAR_ERROR_MESSAGE = 'CLEAR_ERROR_MESSAGE';
export const SET_INITIAL_LOADING = 'SET_INITIAL_LOADING';
export const CHANGE_LOADING = 'CHANGE_LOADING';

export const validateEmail = (email) => {
  return /\S+@\S+\.\S+/.test(String(email).toLowerCase());
};

export const validatePassword = (pwd) => {
  return /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]{8,}$/.test(String(pwd));
};

export const authReducer = (prevState, action) => {
  switch (action.type) {
    case SET_ERROR_MESSAGE:
      return {
        ...prevState,
        errorMessage: action.payload,
      };
    case SIGN_IN:
      return {
        ...prevState,
        ...action.payload,
        errorMessage: '',
      };
    case SIGN_OUT:
      return {
        ...prevState,
        token: null,
      };
    case SET_LOCAL_INFO:
      return {
        ...prevState,
        ...action.payload,
      };
    case CLEAR_ERROR_MESSAGE:
      return {
        ...prevState,
        errorMessage: '',
      };
    case SET_INITIAL_LOADING:
      return {
        ...prevState,
        initialyLoaded: true,
      };
    case CHANGE_LOADING:
      return {
        ...prevState,
        loading: action.payload,
      };
    case CHANGE_REGISTERING:
      return {
        ...prevState,
        registering: action.payload,
      };
    default:
      return prevState;
  }
};

const signup = (dispatch) => ({ email, password, password2 }) => {
  if (!email || !password2 || !password) {
    dispatch({ type: SET_ERROR_MESSAGE, payload: 'Не все поля заполнены!' });
    return false;
  }

  if (!validateEmail(email)) {
    dispatch({ type: SET_ERROR_MESSAGE, payload: 'Email введён неправильно.' });
    return false;
  }

  if (!validatePassword(password)) {
    dispatch({
      type: SET_ERROR_MESSAGE,
      payload: `Пароль введён неправильно. 
Пароль должен содержать минимум:
- 1 цифру;
- 1 букву в нижнем регистре;
- 1 букву в верхнем регистре.
Минимальная длинна пароля - 8 символов.`,
    });
    return false;
  }

  if (password !== password2) {
    dispatch({ type: SET_ERROR_MESSAGE, payload: 'Пароли не совпадают.' });
    return false;
  }

  return new Promise(async (resolve) => {
    dispatch({ type: CHANGE_LOADING, payload: true });
    dispatch({ type: CHANGE_REGISTERING, payload: true });

    try {
      dispatch({ type: CLEAR_ERROR_MESSAGE });
      const {
        data: { token, user },
      } = await ProofAPI.post('/signup', { email, password });

      await AsyncStorage.setItem('token', token);
      await AsyncStorage.setItem('user', JSON.stringify(user));

      resolve();

      Toast.showWithGravity(
        'Вы зарегистрировались!',
        Toast.SHORT,
        Toast.CENTER
      );
      dispatch({ type: SIGN_IN, payload: { token, user } });
    } catch (err) {
      dispatch({ type: SET_ERROR_MESSAGE, payload: 'Ошибка регистрации.' });
    }

    dispatch({ type: CHANGE_LOADING, payload: false });
  });
};

const signin = (dispatch) => async ({ email, password }) => {
  dispatch({ type: CHANGE_LOADING, payload: true });
  try {
    dispatch({ type: CLEAR_ERROR_MESSAGE });
    const {
      data: { token, user },
    } = await ProofAPI.post('/signin', { email, password });
    await AsyncStorage.setItem('token', token);
    await AsyncStorage.setItem('user', JSON.stringify(user));
    dispatch({ type: SIGN_IN, payload: { token, user } });
    Toast.showWithGravity('Вы вошли в аккаунт!', Toast.SHORT, Toast.CENTER);
  } catch (err) {
    dispatch({ type: SET_ERROR_MESSAGE, payload: 'Ошибка входа.' });
  }
  dispatch({ type: CHANGE_LOADING, payload: false });
};

const signout = (dispatch) => async () => {
  dispatch({ type: CHANGE_LOADING, payload: true });
  await AsyncStorage.removeItem('token');
  await AsyncStorage.removeItem('user');
  Toast.showWithGravity('Вы вышли из аккаунта', Toast.SHORT, Toast.CENTER);
  dispatch({ type: SIGN_OUT });
  dispatch({ type: CHANGE_LOADING, payload: false });
};

const clearErrorMessage = (dispatch) => () => {
  dispatch({ type: CLEAR_ERROR_MESSAGE });
};

const tryLocalSignin = (dispatch) => async () => {
  const token = await AsyncStorage.getItem('token');
  const user = await AsyncStorage.getItem('user');

  dispatch({ type: SIGN_IN, payload: { token, user: JSON.parse(user) } });
  dispatch({ type: SET_INITIAL_LOADING });
};

const getLocalInfo = (dispatch) => async () => {
  const myAdvert = await AsyncStorage.getItem('myAdvert');
  dispatch({
    type: SET_LOCAL_INFO,
    payload: { myAdvert: JSON.parse(myAdvert) },
  });
};

const setLocalInfo = (dispatch) => async (myAdvert) => {
  await AsyncStorage.setItem('myAdvert', JSON.stringify(myAdvert));
  dispatch({ type: SET_LOCAL_INFO, payload: { myAdvert } });
};

const changeRegistering = (dispatch) => (payload) => {
  dispatch({ type: CHANGE_REGISTERING, payload });
};

const updateProfile = (dispatch) => async (payload, cb = null) => {
  dispatch({ type: CHANGE_LOADING, payload: true });

  try {
    const { data: user } = await ProofAPI.post('/profile', payload);
    await AsyncStorage.setItem('user', JSON.stringify(user));

    if (cb) {
      cb();
    }

    dispatch({ type: SIGN_IN, payload: { user } });
    Toast.showWithGravity(
      'Данные успешно сохранены!',
      Toast.SHORT,
      Toast.CENTER
    );
  } catch (err) {
    console.error(err);
  }

  dispatch({ type: CHANGE_REGISTERING, payload: false });
  dispatch({ type: CHANGE_LOADING, payload: false });
};

export const { Provider, Context } = createDataContext(
  authReducer,
  {
    signin,
    signup,
    signout,
    clearErrorMessage,
    tryLocalSignin,
    getLocalInfo,
    setLocalInfo,
    changeRegistering,
    updateProfile,
  },
  initialState
);
