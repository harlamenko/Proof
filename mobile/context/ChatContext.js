import httpClient from "../api/ProofAPI";
import createDataContext from "./createDataContext";
import io from 'socket.io-client';
import { Conversation } from "../models/Conversation";

const ADD_CONVERSATION = 'ADD_CONVERSATION';
const UPDATE_CONVERSATIONS = 'UPDATE_CONVERSATIONS';
const SET_EMPTY_MESSAGE = 'SET_EMPTY_MESSAGE';
const SET_CONVERSATION = 'SET_CONVERSATION';
const CHANGE_LOADING = 'CHANGE_LOADING';
const SET_MESSAGES = 'SET_MESSAGES';

const reducer = (prevState, action) => {
    switch (action.type) {
        case ADD_CONVERSATION:
            return {
                ...prevState,
                conversations: action.payload
            }
        case SET_EMPTY_MESSAGE:
            return {
                ...prevState,
                emptyMessage: action.payload
            }
        case SET_CONVERSATION:
            const prevConv = prevState.currentConversation || {};
            return {
                ...prevState,
                currentConversation: { ...prevConv, ...action.payload }
            }
        case CHANGE_LOADING:
            return {
                ...prevState,
                loading: action.payload
            }
        case SET_MESSAGES:
            return {
                ...prevState,
                messages: action.payload
            }
        case UPDATE_CONVERSATIONS:
            const { conversations } = prevState;
            const { _id, updated_at, last_message } = action.payload
            const c = conversations.find(c => c._id === _id);

            if (c) {
                c.updated_at = updated_at;
                c.last_message = last_message
            } else {
                conversations.push(new Conversation(action.payload));
            }

            conversations.sort((a, b) => b.updated_at - a.updated_at);

            return {
                ...prevState,
                conversations
            }
        default:
            return prevState;
    }
}

const getConversations = dispatch => async () => {
    dispatch({ type: CHANGE_LOADING, payload: true });
    try {
        // TODO добавить кэширование
        // const data = JSON.parse('[{"_id":"5ec453fa7198462087a459d9","advert":{"_id":"5ebfb6df44b6ec09200bd04e","user_id":"5ead94a6ef25f930080cb561","name":"qwer","publication_date":"2020-05-16T09:48:15.093Z","price":1234,"city":"qwert","model_name":"Android SDK built for x86","build_id":"QSR1.190920.001","brand_name":"google","year_class":2013,"os_name":"Android","description":"qwert fghj","photos":[{"_id":"5ebfb6df44b6ec09200bd04d","photo":"blya"}]},"buyer":{"_id":"5eab0126b8121445a47ba7c5","email":"mail1@mail.com"},"seller":{"_id":"5ead94a6ef25f930080cb561","email":"admin"},"last_message":"testing for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my momtesting for my mom","updated_at":"2020-05-21T15:46:07.838Z"}]');
        const { data } = await httpClient.get('/conversations');

        if (data && data.length) {
            dispatch({ type: ADD_CONVERSATION, payload: data.map(el => new Conversation(el)) });
        } else {
            dispatch({ type: SET_EMPTY_MESSAGE, payload: 'Список диалогов пуст.' });
        }
    } catch (err) {
        console.error(err);
    }
    dispatch({ type: CHANGE_LOADING, payload: false });
}

const updateConversations = dispatch => payload => {
    dispatch({ type: UPDATE_CONVERSATIONS, payload });
}

const clearEmptyMessage = dispatch => () => {
    dispatch({ type: SET_EMPTY_MESSAGE, payload: null })
}

const sendMessage = dispatch => async info => {
    try {
        const { data } = await httpClient.post(`/conversation/${info.msg.conversation}`, info);

        if (!data) { return; }

        dispatch({ type: SET_CONVERSATION, payload: { _id: data.conversationId } });
    } catch (err) {
        console.error(err);
    }
}

const tryGetConversation = dispatch => async conv => {
    const { seller, advert, buyer, _id: cid = null } = conv;

    dispatch({ type: CHANGE_LOADING, payload: true });
    dispatch({ type: SET_CONVERSATION, payload: new Conversation(conv) })

    try {
        if (cid) {
            const { data: { messages } } = await httpClient.get(`/conversation/${cid}`);

            dispatch({ type: SET_MESSAGES, payload: messages });
            dispatch({ type: CHANGE_LOADING, payload: false });
            return;
        }

        const { data } = await httpClient.get(`/conversation/${seller}/${advert}/${buyer}`);
        dispatch({ type: SET_CONVERSATION, payload: new Conversation(data) });

        if (!data || !data._id) { return; }

        // TODO добавить кэширование
        const { data: { messages } } = await httpClient.get(`/conversation/${data._id}`);

        dispatch({ type: SET_MESSAGES, payload: messages });
        dispatch({ type: CHANGE_LOADING, payload: false });
    } catch (err) {
        console.error(err);
    }

    dispatch({ type: CHANGE_LOADING, payload: false });
};

const getMessages = dispatch => async cid => {
    // TODO добавить кэширование
    dispatch({ type: CHANGE_LOADING, payload: true });
    try {
        dispatch({ type: SET_MESSAGES, payload: [] });

        const { data: { messages } } = await httpClient.get(`/conversation/${cid}`);

        dispatch({ type: SET_MESSAGES, payload: messages });
    } catch (err) {
        console.error(err);
    }
    dispatch({ type: CHANGE_LOADING, payload: false });
}

export const { Provider, Context } = createDataContext(
    reducer,
    {
        getConversations,
        clearEmptyMessage,
        sendMessage,
        getMessages,
        tryGetConversation,
        updateConversations
    },
    {
        conversations: null,
        emptyMessage: null,
        currentConversation: null,
        messages: null,
        loading: false
    }
);