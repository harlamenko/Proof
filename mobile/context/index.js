import { Provider as AuthProvider, Context as AuthContext } from './AuthContext';
import { Provider as AdvertsProvider, Context as AdvertsContext } from './AdvertsContext';
import { Provider as ChatProvider, Context as ChatContext } from './ChatContext';

export { AuthProvider, AuthContext, AdvertsProvider, AdvertsContext, ChatProvider, ChatContext };
