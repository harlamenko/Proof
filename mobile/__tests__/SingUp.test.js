jest.mock('react-native-simple-toast', () => ({
  SHORT: jest.fn(),
}));
import React from 'react';
import { cleanup, render } from 'react-native-testing-library';
import { AuthProvider } from '../context';
import SignUp from '../screens/SingUp';

const TestComponent = () => (
  <AuthProvider>
    <SignUp testId="SignUp" />
  </AuthProvider>
);

describe('SingUp', () => {
  afterEach(cleanup);

  it('Должен рендерится', () => {
    expect(render(<TestComponent />)).not.toBeNull();
  });

  it('Должен рендерится корректно', () => {
    const SignInScreen = render(<TestComponent />).toJSON();
    expect(SignInScreen).toMatchSnapshot();
  });
});
