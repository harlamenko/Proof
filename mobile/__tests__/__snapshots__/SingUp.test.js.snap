// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingUp Должен рендерится корректно 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 35,
        "marginBottom": 22,
        "marginHorizontal": 10,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Зарегистрироваться
  </Text>
  <View
    style={
      Object {
        "marginVertical": 8,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Email
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="admin"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Пароль
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="admin"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 8,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Повторите пароль
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="password2"
        underlineColorAndroid="transparent"
        value="admin"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
        "marginHorizontal": 10,
        "marginTop": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="submitBtn"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Зарегистрироваться
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
        "marginBottom": -40,
        "marginTop": 40,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2089dc",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          ВХОД
        </Text>
      </View>
    </View>
  </View>
</View>
`;
